### PR TITLE
wasm mvp node with web GUI

### PR DESCRIPTION
**Pull request by hanabi1224**
**Summary of changes**
Changes introduced in this pull request:
- websocket proxy node
- wasm node with web GUI
- docs / instructions
- CI job
- Instructions of building and running the demo locally in readme.md(s)

proxy node features:
Done:
- retrieve node status from mina network
TODO:
- retrieve blocks

wasm node features:
Done:
- show node status on web GUI
TODO:
- all block related logic (to be broken down)

**Reference issue to close (if applicable)**

Closes  https://github.com/ChainSafe/mina-rs/issues/177

**Other information and links**

### CODE REVIEW

File: apps/wasm/mvp/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+pub fn wasm_test() -> bool {
+    log_str("wasm_test");
+    true
+}
+
+#[wasm_bindgen]
+pub async fn wasm_test_async() -> bool {
+    log_str("wasm_test_async");
+    true
+}
```
Comment (willemolding) [inline apps/wasm/mvp/src/lib.rs:38]: Want to keep these?

File: apps/wasm/mvp/readme.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  cd apps/mvp-proxy
+  go mod vendor
+  go run .
+  ```
+
+- Build and Run
+
+  ```
+  cd apps/wasm/mvp
+  yarn
```
Comment (willemolding) [inline apps/wasm/mvp/readme.md:29]: I had issues here with incompatible node versions. Might be worth mentioning that it required node 16.0 or later

File: apps/wasm/mvp/readme.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
+# Prerequisites
```
Comment (willemolding) [inline apps/wasm/mvp/readme.md:1]: It would be cool to have an overall description of the architecture of this demo and a high level of how it works

**Review by willemolding — COMMENTED** [review]
This is amazing! I'm not really comfortable reviewing the Golang code, that might be best left to @noot. Is there a possibility that the proxy could be written in Rust instead or is the Rust version of libp2p not up to the job?

File: apps/wasm/mvp/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+#[wasm_bindgen]
+pub async fn connect(addr: String) -> bool {
+    connect_async(&addr).await.unwrap()
+}
+
+async fn connect_async(addr: &str) -> anyhow::Result<bool> {
+    let js_promise = js_sys::Promise::resolve(&42.into());
+    let js_future: wasm_bindgen_futures::JsFuture = js_promise.into();
+    let js_val = js_future.await.unwrap();
+    log_string(format!("js_val: {:?}", js_val));
```
Comment (willemolding) [inline apps/wasm/mvp/src/lib.rs:56]: Also this can be dropped right?

Comment (hanabi1224) [reply apps/wasm/mvp/src/lib.rs]: Removed

File: apps/wasm/mvp/src/lib.rs (hunk 1 above)
Comment (hanabi1224) [reply apps/wasm/mvp/src/lib.rs]: Removed.

File: apps/wasm/mvp/readme.md (hunk 3 above)
Comment (hanabi1224) [reply apps/wasm/mvp/readme.md:7]: That's good suggestion, a UML diagram added 
arch-uml

File: apps/wasm/mvp/readme.md (hunk 2 above)
Comment (hanabi1224) [reply apps/wasm/mvp/readme.md:35]: Thx for pointing it out, I haven't tried node  version 14 and prior, a little surprised to know 14 not working. Will update the doc

**Review by willemolding — APPROVED** [review]
Approved pending review of the Go code

File: apps/wasm/mvp/patch.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+content += `
+*/
+module.exports.init = async function (stream) {
+	module.exports.__wasm = wasm;
+    const wasmInstance = await (await WebAssembly.instantiateStreaming(stream, imports)).instance;
+    wasm = wasmInstance.exports;
+    return wasm
+}
+`
+// .replace('const bytes = ', 'const bytes = require("wasm_bg")\nconsole.log("reloading wasm")\n// const bytes =')
```
Comment (creativcoder) [inline apps/wasm/mvp/patch.js:19]: can we remove these?

File: apps/wasm/mvp/readme.md (hunk 3 above)
Comment (creativcoder) [reply apps/wasm/mvp/readme.md:7]: This is really helpful

**Review by creativcoder — APPROVED** [review]
Great progress here :rocket:

File: apps/wasm/mvp/patch.js (hunk 5 above)
Comment (hanabi1224) [reply apps/wasm/mvp/patch.js]: Removed

**Pull request merged**